R/zchunk_LA131.enduse.R

```diff
@@ -35,58 +36,189 @@ module_energy_LA131.enduse_DISABLED <- function(command, ...) {
+    L1011.en_bal_EJ_R_Si_Fi_Yh %>%
+      filter(sector %in% enduse_sector_aggregation$sector) %>%
+      filter(fuel != "electricity") %>%
+      bind_rows(Enduse_elect_scaled) ->
+      Enduse_elect_scaled_heat_unscaled # still need to scale heat
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # HEAT SCALING
+    # Total delivered heat = output of district heat sector + secondary (heat) output of electric sector
+    L124.out_EJ_R_heatfromelec_F_Yh %>%
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value = sum(value)) ->
+      Heatfromelect
+
+    L124.out_EJ_R_heat_F_Yh %>%
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value = sum(value)) %>%
+      left_join_error_no_match(Heatfromelect, by = c("GCAM_region_ID", "year")) %>%
+      mutate(value = value.x + value.y) %>%
+      select(-value.x, -value.y) ->
+      Enduse_heat
+
+    # Subset the end use sectors and aggregate by fuel. Only in regions where heat is modeled as a separate fuel.
+    A_regions %>%
+      filter(heat == "1") %>%
+      select(GCAM_region_ID) %>%
+      .$GCAM_region_ID ->
+      GCAM_region_ID_heat
+
+    Enduse_elect_scaled_heat_unscaled %>%
+      filter(fuel == "heat", GCAM_region_ID %in% GCAM_region_ID_heat) %>%
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value = sum(value)) ->
+      Enduse_heat_unscaled
+
+    # Calculate the scalers required to balance district heat production and consumption within each region
+    Enduse_heat %>%
+      left_join_error_no_match(Enduse_heat_unscaled, by = c("GCAM_region_ID", "year")) %>%
+      mutate(value = value.x / value.y) %>%
+      mutate(value = if_else(is.na(value),0,value)) %>%
+      select(-value.x, -value.y) ->
+      Enduse_heat_scaler
+
+    # Multiply the district heat scalers by the original estimates of district heat consumption by end use sectors
+    Enduse_elect_scaled_heat_unscaled %>%
+      filter(fuel == "heat", GCAM_region_ID %in% GCAM_region_ID_heat) %>%
+      left_join_error_no_match(Enduse_heat_scaler, by = c("GCAM_region_ID", "year")) %>%
+      mutate(value = value.x * value.y) %>%
+      select(-value.x, -value.y) ->
+      Enduse_heat_scaled
+
+    # Replace unscaled estimates of end use sector heat consumption in full table
+    Enduse_elect_scaled_heat_unscaled %>%
+      mutate(fuel = replace(fuel, GCAM_region_ID %in% GCAM_region_ID_heat & fuel == "heat", "heat2")) %>%
+      filter(fuel != "heat2") %>%
+      bind_rows(Enduse_heat_scaled) %>%
```

Comment:
heat2 is a little confusing. Maybe you could use "heat_scaled" since you are filtering that out to bind with heat scaled.